packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -0,0 +1,310 @@
+     * 1. If we have valid access token - use it
+     * 2. If we have valid refresh token - refresh token flow
+     * 3. PKCE flow
+     *
+     * Only call this from the "ready" function, so "ready" can manage the pending state.
+     *
+     * @internal
+     */
+    private async init() {
+        if (!this.isTokenExpired(this.get('access_token'))) {
+            return this.data
+        }
+
+        const refreshToken = this.get('refresh_token_registered') || this.get('refresh_token_guest')
+
+        if (refreshToken) {
+            const res = await helpers.refreshAccessToken(this.client, {refreshToken})
+            this.handleTokenResponse(res, true)
+            return this.data
+        }
+
+        const res = await helpers.loginGuestUser(this.client, {redirectURI: this.redirectURI})
+        this.handleTokenResponse(res, true)
+
+        return this.data
+    }
+
+    private handleTokenResponse(res: ShopperLoginTypes.TokenResponse, isGuest: boolean) {
+        this.set('access_token', `Bearer ${res.access_token}`)
+        this.set('customer_id', res.customer_id)
+        this.set('enc_user_id', res.enc_user_id)
+        this.set('expires_in', `${res.expires_in}`)
+        this.set('id_token', res.id_token)
+        this.set('idp_access_token', res.idp_access_token)
+        this.set('token_type', res.token_type)
+        this.set('usid', res.usid)
+
+        const refreshTokenKey = isGuest ? 'refresh_token_guest' : 'refresh_token_registered'
+        this.set(refreshTokenKey, res.refresh_token, {
+            expires: this.REFRESH_TOKEN_EXPIRATION_DAYS
+        })
+    }
+
+    /**
+     * The ready function returns a promise, signaling whether or not
+     * the access token is avaliable. If access token is not avaliable
+     * this method will try to re-initialize.
+     *
+     * We use this method to block those commerce api calls that
+     * requires an access token.
+     */
+    async ready() {
+        if (!this.pending) {
+            this.pending = this.init()
+        }
+        return this.pending
+    }
+
+    async loginGuestUser() {
+        const redirectURI = this.redirectURI
+        const usid = this.get('usid')
+        const request = async () => {
+            const res = await helpers.loginGuestUser(this.client, {
+                redirectURI,
+                ...(usid && {usid})
+            })
+            this.handleTokenResponse(res, true)
+            return this.data
+        }
+        this.pending = request()
+        return this.pending
+    }
+
+    async loginRegisteredUserB2C(credentials: Parameters<Helpers['loginRegisteredUserB2C']>[1]) {
+        const redirectURI = this.redirectURI
+        const usid = this.get('usid')
+        const request = async () => {
+            const res = await helpers.loginRegisteredUserB2C(this.client, credentials, {
+                redirectURI,
+                ...(usid && {usid})
+            })
+            this.handleTokenResponse(res, true)
```

Comment:
Shouldn't this be false since this is a registered user login?

Reply:
thanks for catching the bug, fixed.